### ISSUE TITLE
BucketAlreadyOwnedByYou Error while creating buckets, FS and DB shows bucket entry

### ISSUE THREAD

**Issue by akmithal** [noobaa/noobaa-core#6885]
### Environment info
- NooBaa Version: **VERSION**
- Platform: OCP 4.9

### Actual behavior
1. Bucket creation failed with error - BucketAlreadyOwnedByYou though buckets got created on filesystem and DB

### Expected behavior
1. No error

### Steps to reproduce

Hi @romayalon , I attempted large no. of buckets creation and I was able to recreate this:

```
make_bucket failed: s3://bucket-1793 An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it. You get this error in all AWS regions except US East (N. Virginia) region, us-east-1. In us-east-1 region, you will get 200 OK, but it is no-op (if bucket exists it Amazon S3 will not do anything).
...
make_bucket failed: s3://bucket-1794 An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it. You get this error in all AWS regions except US East (N. Virginia) region, us-east-1. In us-east-1 region, you will get 200 OK, but it is no-op (if bucket exists it Amazon S3 will not do anything).
...
make_bucket: bucket-1795
...
make_bucket failed: s3://bucket-1796 An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it. You get this error in all AWS regions except US East (N. Virginia) region, us-east-1. In us-east-1 region, you will get 200 OK, but it is no-op (if bucket exists it Amazon S3 will not do anything).
...
make_bucket: bucket-1797
...
make_bucket: bucket-1798
...
make_bucket failed: s3://bucket-1799 An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it. You get this error in all AWS regions except US East (N. Virginia) region, us-east-1. In us-east-1 region, you will get 200 OK, but it is no-op (if bucket exists it Amazon S3 will not do anything).
...
make_bucket failed: s3://bucket-1800 An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it. You get this error in all AWS regions except US East (N. Virginia) region, us-east-1. In us-east-1 region, you will get 200 OK, but it is no-op (if bucket exists it Amazon S3 will not do anything).
```

When I am checking on filesystem, these entries are  indeed created:

```
[root@hpo-21 ~]# s3m1 ls |grep bucket-179

2022-02-07 05:06:26 bucket-1799
2022-02-07 05:06:26 bucket-1798
2022-02-07 05:06:26 bucket-1797
2022-02-07 05:06:26 bucket-1796
2022-02-07 05:06:26 bucket-1795
2022-02-07 05:06:26 bucket-1794
2022-02-07 05:06:26 bucket-1793
2022-02-07 05:06:26 bucket-1792
2022-02-07 05:06:26 bucket-1791
2022-02-07 05:06:26 bucket-1790
2022-02-07 05:06:26 bucket-179
```

All these buckets are present on the filesystem as well. 

```
sh-4.4# ls -lrt |grep bucket-179
drwxr-x---.  2 1000 users  4096 Feb  7 07:21 bucket-179
drwxr-x---.  2 1000 users  4096 Feb  7 12:56 bucket-1790
drwxr-x---.  2 1000 users  4096 Feb  7 12:57 bucket-1791
drwxr-x---.  2 1000 users  4096 Feb  7 12:58 bucket-1792
drwxr-x---.  2 1000 users  4096 Feb  7 12:59 bucket-1793
drwxr-x---.  2 1000 users  4096 Feb  7 13:00 bucket-1794
drwxr-x---.  2 1000 users  4096 Feb  7 13:01 bucket-1795
drwxr-x---.  2 1000 users  4096 Feb  7 13:02 bucket-1796
drwxr-x---.  2 1000 users  4096 Feb  7 13:02 bucket-1797
drwxr-x---.  2 1000 users  4096 Feb  7 13:03 bucket-1798
drwxr-x---.  2 1000 users  4096 Feb  7 13:04 bucket-1799
```

### More information - Screenshots / Logs / Other output
[must-gather-0702_1.tar.gz](https://github.com/noobaa/noobaa-core/files/8015543/must-gather-0702_1.tar.gz)
[test_07feb.db.zip](https://github.com/noobaa/noobaa-core/files/8015556/test_07feb.db.zip)

**Comment by romayalon** [comment]
Hi @akmithal,
Were you able reproduce it with the new fix of #6856  ?

**Comment by akmithal** [comment]
Thanks @romayalon , This is not reproducible anymore, so closing this bug.
